spec/page_spec.rb

```diff
@@ -0,0 +1,227 @@
+RSpec.describe Fauna::Page do
+  before(:all) do
+    create_test_db
+    @test_class = client.query { create ref('classes'), name: 'page_test' }[:ref]
+
+    index_all = client.query { create ref('indexes'), name: 'page_all', source: @test_class }
+
+    wait_for_active(index_all[:ref])
+
+    @test_index = index_all[:ref]
+
+    @instances = client.query { (1..3).collect { |x| create(@test_class, data: { value: x }) } }
+    @instance_refs = @instances.collect { |instance| instance[:ref] }
+
+    @test_match = Fauna::Query.match(@test_index)
+  end
+
+  after(:all) do
+    destroy_test_db
+  end
+
+  describe '#cursor' do
+    it 'only returns cursors' do
+      before = random_ref
+      after = random_ref
+
+      # This is not a valid configuration. It is only used to test the property
+      page = Fauna::Page.new(client, @test_match, ts: random_number, sources: true, before: before, after: after)
```

Comment:
I can add a helper to do that. I'll also rework this to just take two passes at validating this instead of doing both cursors at the same time.